webpack/global_index.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { addGlobalFill } from 'foremanReact/components/common/Fill/GlobalFill';
+import { registerReducer } from 'foremanReact/common/MountingService';
+import API from 'foremanReact/API';
+import reducers from './ForemanRhCloudReducers';
+import InsightsHostDetailsTab from './InsightsHostDetailsTab';
+import { subscriptionCountListener } from './ForemanInventoryUpload/SubscriptionsPageExtension/SubscriptionsPageExtensionActions';
+import { foremanUrl } from './ForemanRhCloudHelpers';
+
+(async () => {
+  // Register reducers
+  Object.entries(reducers).forEach(([key, reducer]) =>
+    registerReducer(key, reducer)
+  );
+
+  // Register css
+  const {
+    data: { linkElement },
+  } = await API.get(foremanUrl('/foreman_rh_cloud/plugin_css_path'));
+  document
+    .getElementsByTagName('head')[0]
+    .insertAdjacentHTML('beforeend', linkElement);
+
+  const [prefix, mainPath, subPath] = window.location.pathname.split('/');
+  switch (mainPath) {
+    case 'subscriptions':
+      window.tfm.store.observeStore(
+        'katello.subscriptions.itemCount',
+        subscriptionCountListener
+      );
+      break;
+    case 'experimental':
```

Comment:
is this switch case triggering once or on every page transition? 
Scenario:
1. Go to a react based page (i.e Audits) - This switch case catch  nothing
2. Navigate to a hosts details page - This switch case isn't re-triggered
3.  Insights tab isn't registered

If this switch case triggers on every page transition we need to verify that we don't register fills twice.

Reply:
need a bit more investigation, though IMO when it comes to React pages it will get registered just once.